ister_gui.py

```diff
@@ -3158,32 +3264,37 @@ def automatic_install(self):
             {'name': 'contenturl', 'out': '--contenturl={0}'},
             {'name': 'versionurl', 'out': '--versionurl={0}'},
             {'name': 'format', 'out': '--format={0}'}]
-        flags = [item['out'].format(self.args[item['name']])
+        ister_cmd = [item['out'].format(self.args[item['name']])
                  for item in supported if self.args[item['name']] is not None]
         ister_log = '/var/log/ister.log'
-        ister_cmd = [sys.executable,
-                     '/usr/bin/ister.py',
-                     '-t',
-                     '/tmp/template.json']
-        ister_cmd.extend(flags)
-        ister_cmd.extend(['-l', ister_log])
         self.logger.debug(' '.join(ister_cmd))
-        watch = WatchableProcess(ister_cmd)
-        watch.start()
-        current = 0
-        while not watch.done:
-            if len(watch.output) > current:
-                current = len(watch.output)
-                text = '\n'.join(watch.output[-10:])
-                Alert(title, text, block=False).do_alert()
-        Alert(title, text).do_alert()
+        args = ister.handle_options(ister_cmd)
+
+        ister.LOG = ister.logging.getLogger('ister')
+        ister.handle_logging(args.loglevel, ister_log)
+        alertLogger = AlertLoggerHandler(title)
+        alertLogger.setLevel(logging.INFO)
+        ister.LOG.addHandler(alertLogger)
+
+        open(ister_log, 'w').close()
+        fhandler = logging.FileHandler(ister_log)
+        fhandler.setLevel(logging.DEBUG)
+        formatter = logging.Formatter(
+            '%(asctime)s-%(levelname)s: %(message)s')
+        fhandler.setFormatter(formatter)
+        ister.LOG.addHandler(fhandler)
```

Comment:
I feel like this should be something ister handles as it is quite a lot of goo that shouldn't be exposed to the caller. I'll add an issue to the github for this though and I can come back to it later.

Reply:
yes I did by this way in order to not change a lot of the ister.py but it could be handled by ister.py just requesting for a handler and using a default if it not exist

Reply:
I will change this in order to provide handle it in a better way